integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/BridgeSteps.java

```diff
@@ -208,9 +210,36 @@ public void bridgeHasErrorHandlerOfTypeAndParameters(String testBridgeName, Stri
         });
     }
 
+    @And("^the Bridge \"([^\"]*)\" has a polling error handler endpoint$")
+    public void bridgeHasPollingErrorHandlerEndpoint(String testBridgeName) {
+        BridgeContext bridgeContext = context.getBridge(testBridgeName);
+        BridgeResponse response = BridgeResource
+                .getBridgeDetails(context.getManagerToken(), bridgeContext.getId());
+        assertThat(response.getErrorHandler().getType()).isEqualTo(ENDPOINT_ERROR_HANDLER_TYPE);
+        assertThat(BridgeUtils.getOrRetrieveBridgePollingErrorHandlerEndpoint(context, testBridgeName)).matches(ENDPOINT_URL_REGEX);
+    }
+
+    @Then("^the polling error handler endpoint of the Bridge \"([^\"]*)\" contains message \"([^\"]*)\" within (\\d+) (?:minute|minutes)$")
+    public void pollingErrorHandlerContainsMessage(String testBridgeName, String errorMessage, int timeoutMinutes) {
+        String resolvedErrorMessage = ContextResolver.resolveWithScenarioContext(context, errorMessage);
+        BridgeContext bridgeContext = context.getBridge(testBridgeName);
+        String errorHandlerEndpoint = bridgeContext.getErrorHandlerEndpoint();
+        Awaitility.await()
+                .conditionEvaluationListener(new AwaitilityOnTimeOutHandler(() -> BridgeResource
+                        .getBridgeErrorHandlerResponse(context.getManagerToken(), errorHandlerEndpoint).then().log().all()))
+                .atMost(Duration.ofMinutes(timeoutMinutes))
+                .pollInterval(Duration.ofSeconds(1))
+                .untilAsserted(() -> assertThat(BridgeResource.getBridgeErrorHandlerResponse(context.getManagerToken(), errorHandlerEndpoint)
+                        .then()
+                        .log().ifValidationFails()
+                        .statusCode(200)
+                        .extract()
+                        .jsonPath().getString("items[0].payload.data"))
```

Comment:
A nitpick - should the step iterate over list of items to find the proper message?
From the implementation it looks like only first message is considered.

Reply:
though... it may make sense to iterate through the list, so it can be more reusable... ok, changing that
Thanks for the suggestion @sutaakar

Reply:
Checking number of messages could be done as a separate step, tough that is just optional.